crashreports/internal/unix/CrashReportContext.cpp

```diff
@@ -0,0 +1,155 @@
+/*!********************************************************************
+* 
+ Audacity: A Digital Audio Editor
+
+ CrashReportContext.cpp
+
+ Vitaly Sverchinsky
+
+ Some parts of the code are designed to operate while app is crashing,
+ so there may be some restrictions on heap usage. For more information
+ please read Breakpad documentation.
+
+ **********************************************************************/
+#include "CrashReportContext.h"
+
+#include <errno.h>
+#include <map>
+#include <sstream>
+
+#if defined(__APPLE__)
+#include "client/mac/handler/exception_handler.h"
+#else
+#include "client/linux/handler/exception_handler.h"
+#endif
+
+bool SendReport(CrashReportContext* c, const char* minidumpPath)
+{
+    auto pid = fork();
+    if(pid == 0)
+    {
+        if(c->mParameters[0] != 0)
+        {
+            execl(c->mSenderPath, CRASHREPORTER_PROGRAM_NAME, "-a", c->mParameters, "-u", c->mReportURL, minidumpPath, NULL);
+        }
+        else
+        {
+            execl(c->mSenderPath, CRASHREPORTER_PROGRAM_NAME, "-u", c->mReportURL, minidumpPath, NULL);
+        }
+        fprintf(stderr, "Failed to start handler: %s\n", strerror(errno));
+        abort();
+    }
+    return pid != -1;
+}
+
+namespace
+{
+
+    //converts parameters map to a string, so that the Crash Reporting program
+    //would understand them
+    std::string StringifyParameters(const std::map<std::string, std::string>& parameters)
+    {
+        std::stringstream stream;
+
+        std::size_t parameterIndex = 0;
+        std::size_t parametersCount = parameters.size();
+        for (auto& pair : parameters)
+        {
+            stream << pair.first.c_str() << "=\"" << pair.second.c_str() << "\"";
```

Comment:
If this were a more general purpose program, you would escape " characters in the value string, here and in the Windows version of this function too, and in parseArguments() of CrashReporterApp.cpp.

I had to find the use of SetParameters in Audacity.cpp to confirm this doesn't happen.